### PR TITLE
Attributes of Tendrl objects should be watchable

### PR DESCRIPTION
**Pull request by r0h4n**
tendrl-bug-id: tendrl/commons#837
Signed-off-by: Rohan Kanade <rkanade@redhat.com>

### CODE REVIEW

**Comment by r0h4n** [comment]
@shtripat @nthomas-redhat  We need to implement "on_change" for any object that wants to act on changes to its watched attributes (watch_attrs list in defs)

**Comment by shtripat** [comment]
@r0h4n It would be great if you implement one case say for volume status. That would become a reference for all other cases.

File: tendrl/commons/objects/__init__.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def thread_safe_method(self, *args, **kws):
 @six.add_metaclass(abc.ABCMeta)
 class BaseObject(object):
     def __init__(self, *args, **kwargs):
+        self._ttl = None
```
Comment (shtripat) [inline tendrl/commons/objects/__init__.py]: Wouldn't it affect all other objects?

Comment (r0h4n) [reply tendrl/commons/objects/__init__.py]: The variable names are different, all objects use "ttl", its only now we are storing the "ttl" passed to save as self._ttl  (there's an underscore).

So this wont affect any object except the NodeContext for now

Comment (shtripat) [reply tendrl/commons/objects/__init__.py]: ok. otherwise looks fine to me

**Pull request merged**